### PR TITLE
Add markdown formatter

### PR DESCRIPTION
**Pull request by krk**
### Description
Add two new makefile targets, `check-md` and `format-md`. Run `check-md` in GHA.

### Motivation and context
Having an automatic formatter makes editing markdowns more welcoming, as one would run `make format-md` after making edits to use a common style.

### How has this been tested?
```
make format-md && make check-md
```

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: docs/Troubleshooting.md [hunk 1]
```diff
@@ -1,137 +1,140 @@
...
-* You will not see the non-Java frames _preceding_ the Java frames on the
-stack, unless `--cstack vm` is specified.
-For example, if `start_thread` called `JavaMain` and then your Java
-code started running, you will not see the first two frames in the resulting
-stack. On the other hand, you _will_ see non-Java frames (user and kernel)
-invoked by your Java code.
-
-
-* macOS profiling is limited to user space code only.
+- ```
```
Comment (apangin) [inline docs/Troubleshooting.md:3]: Having one style is good, but I must admit that in some cases lists with `*` look better.
For instance, here `-` is blending with backticks.

Comment (krk) [reply docs/Troubleshooting.md:3]: I can try to make `*` the default list token.

Comment (krk) [reply docs/Troubleshooting.md]: Used idiomatic headers here, instead of code blocks in list items.

File: Makefile [hunk 2]
```diff
@@ -115,7 +115,13 @@ ifneq (,$(findstring $(ARCH_TAG),x86 x64 arm64))
 endif
 
 
-.PHONY: all jar release build-test test native clean coverage clean-coverage build-test-java build-test-cpp test-cpp test-java
+.PHONY: all jar release build-test test native clean coverage clean-coverage build-test-java build-test-cpp test-cpp test-java format-md check-md
+
+check-md:
+	prettier -c README.md "docs/**/*.md" ".github/**/*.md"
```
Comment (apangin) [inline Makefile:121]: `.github/*.md` files are also from AWS templates

File: docs/Troubleshooting.md [hunk 3]
```diff
@@ -1,137 +1,123 @@
...
-     How to check: run `lsof -p PID | grep java_pid`  
-     If it lists a socket file, but the file does not exist, then this is exactly
+## perf_event mmap failed: Operation not permitted
+
+Profiler allocates 8kB perf_event buffer for each thread of the target process.
+Make sure `/proc/sys/kernel/perf_event_mlock_kb` value is large enough
+(more than `8 * threads`) when running under unprivileged user. Otherwise, the above message
+will be printed, and no native stack traces will be collected.
+
+## Failed to change credentials to match the target process: Operation not permitted
```
Comment (apangin) [inline docs/Troubleshooting.md:10]: Error messages as titles look... strange. Especially when they are that large. I'd make them at least two levels smaller.

File: README.md [hunk 4]
```diff
@@ -65,25 +67,25 @@ as an interactive `Flame Graph` that can be viewed in a browser.
 
 ## Basic usage
 
-* [Getting Started](docs/GettingStarted.md)
-* [Profiler Options](docs/ProfilerOptions.md)
-* [Profiling Modes](docs/ProfilingModes.md)
-* [Integrating async-profiler](docs/IntegratingAsyncProfiler.md)
-* [Profiling In Container](docs/ProfilingInContainer.md)
+- [Getting Started](docs/GettingStarted.md)
```
Comment (apangin) [inline README.md:70]: If there is a list with a sublist, is prettier smart enough not to use the same delimiter for both?

Comment (krk) [reply README.md:70]: Not sure I follow, sublist will be indented:

```
- Item
  - Subitem
```

- Item
  - Subitem

File: docs/Troubleshooting.md (hunk 3 above)
Comment (krk) [reply docs/Troubleshooting.md:10]: I think it is fair to have error messages as title, in the Troubleshooting page. Made them smaller.

File: Makefile (hunk 2 above)
Comment (krk) [reply Makefile]: reverted these changes.

File: README.md (hunk 4 above)
Comment (apangin) [reply README.md:70]: I mean
```
 - Top Item 1
   * subitem 1.1
   * subitem 1.2
 - Top Item 2
```
I don't want autoformatting to replace `*` with `-` in this case.

File: docs/Troubleshooting.md [hunk 5]
```diff
@@ -1,137 +1,125 @@
...
+the corrupted contents in `/proc/[pid]/maps`. The problem is known to
+occur in a container when running Ubuntu with Linux kernel 5.x.
+This is the OS bug, see <https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1843018>.
+
+### Could not open output file
+
+Output file is written by the target JVM process, not by the profiler script.
+Make sure the path specified in `-f` option is correct and is accessible by the JVM.
+
+- No Java stacks will be collected if `-XX:MaxJavaStackTraceDepth` is zero
```
Comment (apangin) [inline docs/Troubleshooting.md:97]: A title is needed, otherwise it reads as continuation of `Could not open output file` section.

"Limitations" maybe?

**Pull request merged**